app/services/search/postgres/article.rb

```diff
@@ -15,10 +15,10 @@ def self.search_documents(
 
         relation = relation.search_articles(term) if term.present?
 
-        tag_flares = Homepage::FetchTagFlares.call(relation)
-
         relation = sort(relation, sort_by, sort_direction)
 
+        tag_flares = Homepage::FetchTagFlares.call(relation)
```

Comment:
this also fixes a bug, as both queries need to have the same exact `ORDER BY` :D